File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/Wrist.java
package org.firstinspires.ftc.teamcode.robot;

import com.qualcomm.robotcore.hardware.HardwareMap;
import com.qualcomm.robotcore.hardware.Servo;

public class Wrist {
    private final Servo wrist;
    private Position position;

    public Wrist(HardwareMap hardwareMap) {
        wrist = hardwareMap.servo.get("wrist");
        position = Position.INTAKE;
        setPosition(position);
    }

    public enum Position {
        INTAKE(0.9),
        HOVER(0.94),
        GRAB_WALL(0.85),
        SCORE(0.94),
        LINE_UP(0.75);

        private final double position;

        Position(double position) {
            this.position = position;
        }

        public double getPosition() {
            return position;
        }
    }

    public void setPosition(Position position) {
        this.position = position;
        wrist.setPosition(position.getPosition());
    }

    public void setRawPosition(double position) {
        wrist.setPosition(position);
    }

    public double getRawPosition() {
        return wrist.getPosition();
    }

    public Position getPosition() {
        return position;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot/Wrist.kt
package org.firstinspires.ftc.teamcode.Robot

import com.qualcomm.robotcore.hardware.HardwareMap

class Wrist(hardwareMap: HardwareMap) {
    private val wrist = hardwareMap.servo.get("wrist")

    private var position: Position = Position.INTAKE

    init {
        setPosition(position)
    }

    enum class Position(val position: Double) {
        INTAKE(0.9),
        HOVER(0.94),
        GRAB_WALL(0.85),
        SCORE(0.94),
        LINE_UP(0.75),
    }

    fun setPosition(position: Position) {
        this.position = position
        wrist.position = position.position
    }

    fun setRawPosition(position: Double) {
        wrist.position = position
    }

    fun getRawPosition(): Double {
        return wrist.position
    }

    fun getPosition(): Position {
        return position
    }
}
